src/compiler/scala/tools/nsc/transform/CleanUp.scala

```diff
@@ -108,7 +108,7 @@ abstract class CleanUp extends Statics with Transform with ast.TreeDSL {
               if (method ne null)
                 return method
               else {
-                method = ScalaRunTime.ensureAccessible(forReceiver.getMethod("xyz", methodCache.parameterTypes()))
+                method = ScalaRunTime.lookupStructuralMethod(forReceiver, "xyz", methodCache.parameterTypes())
```

Comment:
How about emitting:

```scala
method = try { ScalaRunTime.ensureAccessible(forReceiver.getMethod("xyz", methodCache.parameterTypes())) 
 } catch {
  case e: NoSuchMethodException => try { lookupStructuralMethod(...) } catch { case _: Throwable => throw e }
```

To profit from the bug fix, you'd need to the latest standard library available at runtime, but structural calls that used to work will continue to work with the old library.